[Juliet Teoh] iP

JulietTeoh opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

JulietTeoh committed:
Finish level 1: Greet, Echo, Exit
Duke bot can repeat what a user says

JulietTeoh committed:
added an abstract class named "task", which all tasks will inherit from

JulietTeoh committed:
Level 2: Added the ability to store whatever text entered by the user in DUKE

JulietTeoh committed:
Renamed Task to Manager
Echo to Echo Manager
and changed some variable names to better represent the system I am modelling

JulietTeoh committed:
Implemented a class named task, which is the user's task, and ensure that it still works with task 2

JulietTeoh committed:
Task 3: implemented the requirements for task 3 using task manager and task

JulietTeoh committed:
Added Deadline, which is an extension of Task

JulietTeoh committed:
Added Todo, which is another extension of Task

JulietTeoh committed:
Added Event, which is yet another extension of Task

JulietTeoh committed:
Task 4: added support for todo, event and deadlines

JulietTeoh committed:
implemented some exceptions for event, todo and deadline to deal with inaccurate user input

JulietTeoh committed:
shifted the handling of "done" related user input to ListManager, handled exception that can arise from the user during the "done" operation

JulietTeoh committed:
changed the number of "__" to match that of the cs2103t project website

JulietTeoh committed:
updated test that will test the todo, event and deadline features and well as how exception related to these features wills be handled

JulietTeoh committed:
Task 4 completed - able to handle all possible errors in the current version of duke

JulietTeoh committed:
Add deletion feature for Duke

can delete tasks in the list using the command "delete x", where x is a number representing a task in the list

JulietTeoh committed:
updated test to test the deletion feature

JulietTeoh committed:
Add 2 boolean function, checkStringStartingEquals and checkStringEquals, to check userInput.

checkStringStartingEqual checks if the first n characters of String is equal to a given String
checkStringEquals checks if the user input is equal to a given String or equal to a given String + " "
This is to avoid repetitive and messy code.

JulietTeoh committed:
Update automated text ui files

JulietTeoh committed:
Improved the handling of user input

Moved the handling of user input from Duke to ListManager using a new method, handleAllUserInput
Added the padSpaces method to Manager abstract class to help the generating to Strings

JulietTeoh committed:
Catch exception where user does not put a space after a task. eg "todosleep"

JulietTeoh committed:
Level 7 - Save tasks onto hard disk

Changed Task, Todo, Event, Deadline to allow these classes to output the format to be saved in the datafile, and create tasks from parsed input from the datafile

Added feature that will create data folder and dukedata.txt if they dont exist

Added feature that will save the tasks in the datafile automatically when the task list changes

Added feature that will load data from the datafile, restore the state of the task list in the previous session

JulietTeoh committed:
Change default input in dukedata.txt for testing purposes

JulietTeoh committed:
Update testing files to ensure previous features are not lost

JulietTeoh committed:
Level 8 - added date and time handling for deadline tasks, in the dd-mm-yyyy format

JulietTeoh committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Level 8 - added date and time handling for deadline tasks, in the dd-mm-yyyy format

JulietTeoh committed:
Add support for deadline dates of the format yyyy-mm-dd

JulietTeoh committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Add support for deadline dates of the format yyyy-mm-dd

JulietTeoh committed:
Level 8 - Merged the save feature from level 7 with the date feature of level 8

Rectified the Deadline class such that it handles dates properly when loading saved data
Handled exception in ListManager and Duke where the dukedata is not in the correct format for it to become a task

JulietTeoh committed:
Reformat Code Structure by splitting Task Manager into Storage, TaskList, Parser, Ui.

Removed ListManager and renamed EchoManager to Echo.

JulietTeoh committed:
Achieve stretch goal of having commands that can be executed generated from the Parser

JulietTeoh committed:
Reorganised class files into 4 packages under Duke, command, exceptions, task and maincomponents.

duke.unused contains old code that is now defunct, like echo

JulietTeoh committed:
Add 2 J-unit test, one that tests the Deadline task and the other tests the Event task

JulietTeoh committed:
Pre-steps to generate .jar file

JulietTeoh committed:
Implement Find feature, where a user can input a string and get back all the tasks that contain that string

JulietTeoh committed:
Change all files such that it adheres to checkStyle format given

JulietTeoh committed:
Add JavaDocs to some of the classes and methods

JulietTeoh committed:
Merge branch 'branch-Level-9' and resolved conflicts.

* branch-Level-9:
  Implement Find feature, where a user can input a string and get back all the tasks that contain that string

# Conflicts:
#	src/duke/maincomponents/Parser.java

JulietTeoh committed:
Merge branch 'branch-A-JavaDoc' and resolved all conflicts

* branch-A-JavaDoc:
  Add JavaDocs to some of the classes and methods

# Conflicts:
#	src/duke/command/DeadlineCommand.java
#	src/duke/command/DoneCommand.java
#	src/duke/command/ShowListCommand.java
#	src/duke/exceptions/DukeException.java
#	src/duke/maincomponents/TaskList.java
#	src/duke/task/Task.java
#	src/main/java/Duke.java

JulietTeoh committed:
Merge branch 'add-gradle-support'

Moved some files as a result of adding gradle support

JulietTeoh committed:
Move duke files into main.java package

cnlinh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,69 @@
+ * Main class for Duke that handles the execution of its functions
+ */
 public class Duke {
+    private static final String FILENAME = "dukedata.txt";
```
You may want to change the constant var name to be FILE_NAME and FOLDER_NAME.

cnlinh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,69 @@
 public class Duke {
+    private static final String FILENAME = "dukedata.txt";
+    private static final String FOLDERNAME = "data";
+    private static final String PATH = FOLDERNAME + "/" + FILENAME;
```
I think relative file path is prefered

JulietTeoh replied:
yes that would be a better name for the variable

cnlinh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,69 @@
+        boolean isExit = false;
+        String userInput = dukeUi.readCommand();
+
+        while ( !isExit) {
```
Redundant empty space between ( and !isExit?

JulietTeoh replied:
Thanks for catching that error!

cnlinh commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,41 @@
+ * Deadline Command, which adds a deadline task to Duke's TaskList when executed
+ */
+public class DeadlineCommand implements Command {
+    private ArrayList<String> eventDescription;
```
You may want to change to eventDescriptions.

cnlinh commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,32 @@
+
+    @Override
+    public void execute(TaskList dukeTaskList, Ui dukeUi, Storage dukeStorage) {
+        ArrayList<Task> currentTaskList = dukeTaskList.getCurrentTaskList();
```
currentTasks and newTasks?

cnlinh commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,32 @@
+        ArrayList<Task> currentTaskList = dukeTaskList.getCurrentTaskList();
+        ArrayList<Task> newTaskList = new ArrayList<>();
+
+        for (int i = 0; i < currentTaskList.size(); i++) {
```
You may consider using an advanced for loop here.

JulietTeoh replied:
yes i agree

cnlinh commented on src/main/java/duke/maincomponents/Storage.java:
```diff
@@ -0,0 +1,92 @@
+            File dukeData = new File(pathOfDataFile);
+            Scanner dataReader = new Scanner(dukeData);
+            int tempNumberOfTasks = 0;
+            ArrayList<Task> tempArray = new ArrayList<>();
```
You may want to change the var name to tasks

cnlinh commented on src/main/java/duke/maincomponents/TaskList.java:
```diff
@@ -0,0 +1,133 @@
+ * TaskList class, which stores all the Tasks in the list
+ */
+public class TaskList {
+    private ArrayList<Task> TaskArray;
```
taskArray or simply tasks?

cnlinh commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,61 @@
+// import java.time.LocalDateTime;
+
+public class Deadline extends Task {
+    private final DateTimeFormatter formatone = DateTimeFormatter.ofPattern("d-M-uuuu");
```
FORMAT_ONE and FORMAT_TWO?

cnlinh reviewed (commented):
Overall the code is well-organized and readable but there are some coding standard issues.

CharlesLee01 commented on src/main/java/duke/maincomponents/Parser.java:
```diff
@@ -0,0 +1,203 @@
+
+public class Parser {
+
+    public Parser(){
+    }
```
I don't need a constructor for Parser especially when it does nothing.

CharlesLee01 commented on src/main/java/duke/maincomponents/Parser.java:
```diff
@@ -0,0 +1,203 @@
+        return userInput.equals("bye");
+    }
+
+    public Command parse(String userInput) throws DukeException {
+        try {
+            if (this.equalsToList(userInput)) {
+                return new ShowListCommand();
+
+            } else if (this.equalsToDone(userInput)) {
+                int taskDoneInt = this.parseDoneCommand(userInput);
+                return new DoneCommand(taskDoneInt);
+
+            } else if (this.equalsToDelete(userInput)) {
+                int taskDeleteInt = this.parseDeleteCommand(userInput);
+                return new DeleteCommand(taskDeleteInt);
+
+            } else if (this.equalsToToDo(userInput)) {
+                String toDoDescription = this.parseToDoCommand(userInput);
+                return new ToDoCommand(toDoDescription);
+            } else if (this.equalsToEvent(userInput)) {
+                ArrayList<String> eventDescription = this.parseEventCommand(userInput);
+                return new EventCommand(eventDescription);
+            } else if (this.equalsToDeadline(userInput)) {
+                ArrayList<String> eventDescription = this.parseDeadlineCommand(userInput);
+                return new DeadlineCommand(eventDescription);
+            } else if (this.equalsToFind(userInput)) {
+                String stringToFind = this.parseFindCommand(userInput);
+                return new FindCommand(stringToFind);
+            } else {
+                throw new DukeException("I'm sorry, but I don't know what that means");
+            }
+
+        } catch (DukeException e) {
+            throw e;
+        }
+    }
```
I think you can remove the lines before each else if statements.

Example:

if() {
    return;
} else {
    return;
}

JulietTeoh replied:
yes your recommendation would definitely make my code cleaner

CharlesLee01 commented on src/main/java/duke/maincomponents/Storage.java:
```diff
@@ -0,0 +1,92 @@
+public class Storage {
+    private String pathOfDataFile;
+
+    public Storage(String filePath, String folderName) throws DukeException {
+        pathOfDataFile = filePath;
+        try {
+            File folder = new File(folderName);
+            if (folder.mkdir()) {
+                //System.out.println("Folder created: " + folder.getName());
+            } else {
+                //System.out.println("Folder already exists.");
+            }
+            File dataFile = new File(filePath);
+            if (dataFile.createNewFile()) {
+                //System.out.println("File created: " + dataFile.getName());
+            } else {
+                //System.out.println("File already exists.");
+            }
+        } catch (IOException e) {
+            throw new DukeException("Error occured during file or folder creation");
+        }
+    }
```
Instead of commenting out the codes that is not required, removing them will be better.

JulietTeoh replied:
i agree with u

CharlesLee01 reviewed (commented):
Overall, good work with just some minor coding violations.

JulietTeoh committed:
Clean up code in accordance to comments given by peers

JulietTeoh committed:
Add gradle to project and use it to automate tests and build

JulietTeoh committed:
Add checkstyle for project and fix some checkstyle violations

JulietTeoh committed:
Fix some checkstyle errors so less error messages pop up during build

schping commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,41 @@
+    private ArrayList<String> eventDescription;
+
+    /**
+     * Constuct a DeadlineCommand object
+     * @param a Arraylist where index 0 contains the description of the deadline and index 1
```
Perhaps you could include an empty line between description and parameter section to comply with the coding standards?

schping commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,41 @@
+    }
+
+    /**
+     * Executes the deadline command, which adds a deadline task to Duke's TaskList
+     * @param dukeTaskList give dukeTaskList
```
Perhaps you could include an empty line between description and parameter section to comply with the coding standards?

schping commented on src/main/java/duke/command/DoneCommand.java:
```diff
@@ -0,0 +1,38 @@
+        taskDoneInt = i;
+    }
+
+    /**
+     * Executes the Done command, which marks given task as done
```
Perhaps you could include an empty line between description and parameter section to comply with the coding standards? I noticed this in several other places as well.

schping commented on src/main/java/duke/maincomponents/Parser.java:
```diff
@@ -0,0 +1,197 @@
+        return checkStringStartingEquals(userInput, "done");
+    }
+
+    public int parseDoneCommand(String userInput) throws DukeException {
```
Perhaps you could provide a descriptive header for this method, as well as other public methods?

schping commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,25 @@
+import duke.maincomponents.TaskList;
+import duke.maincomponents.Ui;
+import duke.task.Task;
+
```
Perhaps you could include a descriptive header for public classes?

schping reviewed (commented):
Overall I found your work to be very organized and readable! Was quite difficult to spot coding violations aside from those mentioned, so good work on that 👍

JulietTeoh committed:
Create first working gui that has the same features as Duke on the command line

JulietTeoh committed:
Add visual enhancement to UI

Cat will cry if the user enters an input with error, cat will be cute if the user enters a legit input

JulietTeoh committed:
Improve UI such that it stops cutting for text if the todo list is long

JulietTeoh committed:
Minor tweaks such that fat jar file can be made

JulietTeoh committed:
add assertions to code

JulietTeoh committed:
Improve code quality of source code

JulietTeoh committed:
Merge pull request #1 from JulietTeoh/branch-A-Assertions

merge branch which add assertions to code to main branch

JulietTeoh committed:
Merge pull request #2 from JulietTeoh/branch-A-CodeQuality

merge branch that improve code quality of source code to main branch

JulietTeoh committed:
Merge branch 'master' of https://github.com/JulietTeoh/ip

* 'master' of https://github.com/JulietTeoh/ip:
  Improve code quality of source code
  add assertions to code

JulietTeoh committed:
Add feature C-Update
User can now update tasks without having to delete it first using tags /de, /by and /at
/de is the general description for all tasks
/by is only for deadlines and must be a date
/at is only for events

JulietTeoh committed:
Fix checkstyle errors and improve readability of code

JulietTeoh committed:
Resolve most checkstyle errors and remove unneeded code

JulietTeoh committed:
Give credit to reused code for GUI

JulietTeoh committed:
Add picture of UI

JulietTeoh committed:
Change the picture

JulietTeoh committed:
Fix bug where date did not update during update

JulietTeoh committed:
Create user guide and upload picture

JulietTeoh committed:
change jar file name

JulietTeoh committed:
Set theme jekyll-theme-minimal
